front/src/applications/operationalStudies/components/Scenario/ScenarioContent.tsx

```diff
@@ -217,7 +200,10 @@ const ScenarioContent = ({
               )}
               {isMacro ? (
                 <div className={cx(collapsedTimetable ? 'macro-container' : 'h-100')}>
-                  <NGE dto={ngeDto} onOperation={handleNGEOperation} />
+                  {ngeProps.type === 'loading' && <LoaderFill />}
+                  {ngeProps.type === 'ready' && ngeProps.data && (
```

Comment:
This delays loading the NGE webapp until we've got the DTO ready. This regresses performance.

Reply:
Agree, but it also indicates to the user that something is loading.
If I don't do that, I will have the NGE interface (empty but usable), and then a few second later, I will see the data loaded in NGE and chart displayed. The XP is really not good.

The best solution will be to add a loader feature directly in the NGE component (a transparent fill-loader on the NGE app while it is not initialized or data loaded).

Reply:
I think we could show the loader on top of NGE instead. Or hide NGE with `display: none` to let it load offscreen.

Reply:
Bumping this - I don't think we should delay NGE webapp initialization.

Reply:
This has been fixed by changing the code to : 
```
{ngeProps.type === 'loading' && <LoaderFill />}
<NGE
  dto={ngeProps.type === 'ready' && ngeProps.data ? ngeProps.data.dto : undefined}
  onOperation={
    ngeProps.type === 'ready' && ngeProps.data
      ? ngeProps.data?.listener
      : undefined
  }
/>
```